ARCHITECTURE.md

```diff
@@ -0,0 +1,54 @@
+2. **Broad and open community:** Foster a large and powerful community to drive the project’s adoption and direction.
+3. **Composability**: Presto is designed for lakes and lakehouses.  As such, it is engineered to fit cohesively in an ecosystem of other tools, and where possible it strives to integrate with industry standard specifications and libraries that help accomplish the user goals above.
+4. **Coding excellence**: Presto strives to maintain a high bar for contributions.
+
+## Current Architecture
+
+Presto follows a distributed system model with a coordinator and multiple worker nodes. [See Presto Concepts for more information.](https://prestodb.io/docs/0.285.1/overview/concepts.html#server-types)
+
+Long term initiatives in Presto [have corresponding project boards](https://github.com/prestodb/presto/projects?query=is%3Aopen).  Current progress can be roughly understood by looking at the project boards.
+
+## Long Term Vision
+
+### Vertical integration in data lakes
+
+Presto aims to be the top performing system for data lakes.  The top priority for the project is to move fully onto a native evaluation engine, particularly Velox, with the same expectations outlined in the vision for users, particularly user-friendliness and connectivity.
+
+Motivations for this effort are numerous:
+
+1. It is expected to be too difficult to achieve Warehouse performance using a Java evaluation engine.  Some optimizations, such as vectorized execution, in particular require flexibility and precision which is difficult to achieve in Java.
+2. The Presto community is delegating evaluation to a different community, thereby focusing efforts on improved usability.  We believe that shared components can offer network benefits that closed systems may find difficult to achieve, and such a division of efforts will result in a faster and more robust system overall.
+
+### State-of-the-art Optimizer
+
+Presto’s optimizer can learn from long established techniques in more mature systems to provide better plans by default for users.  We are working on bringing these mature optimization techniques into Presto.
+
+### Modularity
+
+Originally, the way that Presto interacted with other systems was unique and proprietary to Presto.  With libraries like Velox and DataFusion standardizing execution, Ibis standardizing data frames, Substrait standardizing an IR, and Arrow standardizing data exchange, there is the possibility of a future that emphasizes more interoperability with other data infrastructure.  We believe that if Presto does not adapt to this future, it stands to be left behind.  Moreover, we are first movers in this area with our early usage of Velox.  As a matter of principle and design, we are aligned in the direction of interoperability, and believe this can be both a differentiator for the project, and allow the project to focus more exclusively on features that make users happy.
```

Comment:
```suggestion
Originally, the way that Presto interacted with other systems was unique and proprietary.  Today, libraries like Velox and DataFusion are standardizing execution, Ibis is standardizing data frames, Substrait is standardizing an IR, and Arrow is standardizing data exchange. These standardizations point to a future that emphasizes interoperability with other data infrastructure. We believe that if Presto does not adapt to this future, it could be left behind.  

In principle and design, we are aligned in the direction of interoperability. We are first movers in this area with our early usage of Velox. We believe that interoperability can be both a differentiator for the project, and allow the project to focus on features that make users happy.
```
A possible revision of this paragraph to two paragraphs, minor suggestions for clarity. 

One consideration is about this: "Substrait [is] standardizing an IR". The abbreviation IR was a stumble for me, I had to look it up in context. Suggest rephrasing or just expanding the acronym.